applications/IgaApplication/custom_processes/mpm_process.cpp

```diff
@@ -0,0 +1,102 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+
+// System includes
+
+
+// External includes
+
+// Project includes
+#include "mpm_process.h"
+
+namespace Kratos
+{
+
+MpmProcess::MpmProcess(
+    Model& rModel,
+    Parameters ThisParameters)
+    : Process()
+    , mrModel(rModel)
+    , mThisParameters(ThisParameters)
+{
+    mThisParameters.ValidateAndAssignDefaults(this->GetDefaultParameters());
+}
+
+void MpmProcess::ExecuteInitializeSolutionStep()
+{
+    std::string model_part_name = mThisParameters["model_part_name"].GetString();
+    ModelPart& r_model_part = mrModel.GetModelPart(model_part_name);
+
+    for (auto& element : r_model_part.Elements()) {
+        auto& r_geometry = element.GetGeometry();
+
+        /// Update location
+        array_1d<double, 3> mpm_location;
+        r_geometry.GlobalCoordinates(mpm_location, 0);
+        array_1d<double, 3> local_coordinates_2 = element.GetGeometry().IntegrationPoints()[0];
+        element.SetValue(MPM_LOCATION, mpm_location);
+    }
+    ResetNodalVariables(r_model_part);
+
+    for (auto& element : r_model_part.Elements()) {
+        const auto& mpm_location = element.GetValue(MPM_LOCATION);
+        auto& r_parent_geometry = element.GetGeometry().GetGeometryParent(0);
+        array_1d<double, 3> local_coordinates_2 = element.GetGeometry().IntegrationPoints()[0];
```

Comment:
Do you need this variable?

Reply:
nope, thanks for pointing it out. That's a relikt of bad debugging ;)